submissions/balancer-design/universal_8to8.md

```diff
@@ -0,0 +1,68 @@
+
+Universal balancers can do what no balancer should be able to: no matter the configuration, they're input-balanced, output-balanced, *and* throughput-unlimited at the same time! This is a set of properties previously thought to be possessed only by a 2-2 balancer, which is just a single splitter. The drawback is that they get *big*.
+
+### The design process
+
+Let's think about how to convert a 1-4 balancer into a 1-3. That is done by looping the extra output back into the first splitter. The same can be done with a 4-4 balancer, which we will use as our base principle:
+
+![Looped 4-4](https://cdn.discordapp.com/attachments/699618487097884723/813449600852819968/unknown.png)
+
+Now, how would we make it so the balancing doesn't get ruined when we block some outputs? Simple, we use priority splitters to auto-loop the items that can't get out. Of course, we need to merge them back with the input too, and set input priority so the main inputs don't slow down when there are items coming:
+
+![Auto-looping 4-4](https://cdn.discordapp.com/attachments/699618487097884723/813449812728217640/unknown.png)
+
+We now discover a major issue. If the looping items try to enter a belt that's already full, they just back up all the way to the split-off point and block the overflow, putting us back to square one:
+
+![Broken auto-looping 4-4](https://cdn.discordapp.com/attachments/699618487097884723/813450086234587156/unknown.png)
+
+To remedy this, we need to make sure every item on the loop can get to every single one of the loop's exits. The way to do this with the least splitters is a secondary balancer:
+
+![Fixed 4-4](https://cdn.discordapp.com/attachments/699618487097884723/813450357639872553/unknown.png)
+
+And we're pretty much done with the 4-4. It can be improved, but more on that later. Now, let's try our hand at something bigger: a universal 8-8. We do the same steps as before, and get this:
+
+![Basic universal 8-8](https://cdn.discordapp.com/attachments/699618487097884723/813450635667832842/unknown.png)
+
+However, there is an arrangement in which this happens:
+
+![Basic 8-8 bottlenecking](https://cdn.discordapp.com/attachments/699618487097884723/813450849481130012/unknown.png)
+
+6 belts are available, but less than 6 are coming out. This is a problem. The standard 8-8 balancer is throughput-limited, meaning the entire thing is as well. Luckily, this is easily fixed by simply replacing the core balancer with a throughput-unlimited one. Notice how the loopback can stay throughput-limited:
```

Comment:
Again, why doesn't the loop-back balancer have to be throughput unlimited? I figure it won't make much of a difference, since it's good enough already. It's valuable to mention it nonetheless for the slower readers.